ArduPlane/quadplane.cpp

```diff
@@ -2576,8 +2589,16 @@ void QuadPlane::vtol_position_controller(void)
     case QPOS_LAND_COMPLETE:
         break;
     }
-    
-    run_z_controller();
+
+    /* tilt rotors should not start running the z controller unless
+     * the tilt is less than tilt max, or we're already running the z
+     * controller */
+    if (suppress_z_controller) {
+        hold_stabilize(0.01);
```

Comment:
This call means that if were in `QPOS_AIRBRAKE` we call `multicopter_attitude_rate_update` twice in one loop. That may not necessarily be a noticeable problem,  but it will do odd stuff to the PIDs. 

We already have the issue with the existing `hold_hover()` calling `run_z_controller()`, that one is my bug, and suffers the same issue of ruining the Z PIDs twice in one loop. 

Maybe it makes more sense to move the various throttle output methods down in to the height control switch statement. I think the key thing is just to make sure we only call one or the other.

Reply:
I've redone this so avoid the double call (checked in gdb) and to also fix the double Z controller call.